src/modules/jcms_rest/src/Plugin/rest/resource/AbstractRestResourceBase.php

```diff
@@ -540,4 +551,31 @@ public function getEntityQueueItem(EntityInterface $node, FieldItemListInterface
     return $item_values;
   }
 
+  /**
+   * Returns the endpoint from the rest resource "canonical" annotation.
+   *
+   * @return string
+   */
+  function getEndpoint(): string {
+    $r = new \ReflectionClass(static::class);
+    $annotation = $r->getDocComment();
+    preg_match("/\"canonical\" = \"(.+)\"/", $annotation, $endpoint);
```

Comment:
`preg_match()` may not match and return false, I'd check and throw an exception if that's not the case.